metricq/types.py

```diff
@@ -159,10 +160,12 @@ def __mul__(self, factor):
     def __str__(self):
         return "{}s".format(self.s)
 
-    def __eq__(self, other: Union["Timedelta", datetime.timedelta]):
+    def __eq__(self, other: object):
         if isinstance(other, datetime.timedelta):
             return self.timedelta == other
-        return self._value == other._value
+        if isinstance(other, Timedelta):
+            return self._value == other._value
+        return NotImplemented
```

Comment:
This is interesting, didn't know about this special value. However, I'm not sure if this is better. We trade static type checking for comparison for the implicit symmetry.

Reply:
Actually this makes the function override safe. See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides

Reply:
Oh, because the parent class  `object` defines `__eq__(self, other: object)`. I see.